script.harmony.control/resources/lib/api/harmony.py

```diff
@@ -0,0 +1,173 @@
+# 0.1.3
+
+import json, websocket
+from . import url
+try:
+    from kodi_six import xbmc
+    MONITOR = xbmc.Monitor()
+except:
+    MONITOR = None
+    import time
+
+
+
+class HubControl:
+
+    def __init__( self, hub_ip, thetimeout=30, delay=250 ):
+        self.HUBIP = hub_ip
+        self.TIMEOUT = thetimeout
+        self.DELAY = delay / 1000
+        self.HUBID = ''
+        self.CONFIG = {}
+        self.COMMANDS = {}
+        self.LOGLINES = []
+        self.WSURL = 'ws://%s:8088/?domain=svcs.myharmony.com&hubId=' % self.HUBIP
+
+
+    def getActivities( self, excludes=None ):
+        self._init_vars()
+        activities = self._get_activities()
+        self.LOGLINES.extend( ['the activities are:', activities] )
+        if excludes:
+            for item in excludes:
+                del activities[item]
+        return activities, self.LOGLINES
+
+
+    def runCommands( self, cmds_str ):
+        self._init_vars()
+        self.LOGLINES.extend( ['the commands passed in are: %s' % cmds_str] )
+        results = []
+        r_cmds = self._parse_cmds( cmds_str )
+        for r_cmd in r_cmds:
+            if r_cmd == 'pause':
+                if MONITOR:
+                    if MONITOR.waitForAbort( self.DELAY ):
+                        return [], self.LOGLINES
+                else:
+                    time.sleep( self.DELAY )
```

Comment:
Why would you not always use a `MONITOR`?

Reply:
I use these APIs in both Kodi and non Kodi projects.  If you look at lines 5 - 10 you'll see that as long as I can import xbmc I use xbmc.Monitor(), so as far as Kodi is concerned I do always use MONITOR. The only instance where time.sleep() is used is if the API is being used outside Kodi.